### ISSUE TITLE
[BUG]: API Keys not working - Stuck pending approval and do not show in manage apps

### ISSUE THREAD

**Issue by katiecoathanger** [etsy/open-api#386]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The create app / API generator is generating API but they do not work, they say 'pending approval' and are not visible on your manage apps section. 

### Steps to reproduce

create app.. get key.. try to use key.. doesn't work.  Try to manage app. doesnt work. 

### Expected behavior

working API Keys

**Comment by Gareth064** [comment]
If you asked for the app to be fully public, and not just private (for now) then your app has to go through an approval process with Etsy. Only do this once your app is ready to be used by your customers, assuming it is not just for you.

**Comment by katiecoathanger** [comment]
> If you asked for the app to be fully public, and not just private (for now) then your app has to go through an approval process with Etsy. Only do this once your app is ready to be used by your customers, assuming it is not just for you.

I requested it to link with a wordpress plugin, not designing my own app.  I've seen a bunch of other people posting across the internet with the same issue, saying they've waited weeks with no change.  Seems to be since they changed to V3.

**Comment by rakeshdobariya1** [comment]
@Gareth064  We are also facing similar issue. We also tried with only selecting option for our self. Still after creating app, status is Pending approval and app is not listed under Manage Your apps. This is blocker for us to move further in development. 

Is there any sandbox available where we can do testing ?

**Comment by Gareth064** [comment]
@rakeshdobariya1 I have only ever seen the approval part when you select that you want your app to have full access, meaning it will be used to connect to other peoples stores and not just your own. If you are just connecting to your own store only then approval isn't needed.

As for a sandbox, annoyingly Etsy doesn't provide one. You have to test against your own live store.

Ps. I don't work for Etsy, I just like answering questions when I think I can help 😂

**Comment by ramo-karahasan-riechardt** [comment]
We face the same issue that we're receiving the "approval" message within etsy with all the API credentials visible, but after re-login the App listing disappears. No matter which type (for me, smaller groupl, ,publich)  is selected

**Comment by jftanner** [comment]
Same issue. I created an app for testing the API, as a 3rd party integration with Make (formerly Integromat) is getting the wrong data. However, the API keys provided don't work and nothing appears under "Manage your apps":
<img width="751" alt="image" src="https://user-images.githubusercontent.com/10095474/156798062-47bbcac0-275e-4a50-9003-d19cd06c0924.png">

I don't need full access, just the provisional API key.

**Comment by Gareth064** [comment]
Wait a minute... this is new
![image](https://user-images.githubusercontent.com/22307223/156798549-b89faf8e-d5fd-4554-91b1-14f08eb41f52.png)
@etsyachristensen are you able to comment on this so called sandbox API?

@jftanner @ramo-karahasan-riechardt @katiecoathanger @rakeshdobariya1 
I can see what you guys are saying... I just created another app, the same way I did with the one I am using now, and it said it was pending. Then when I left that screen and went back to "Manage my apps" that new app wasn't listed. I wonder if this is an oversight on the Etsy devs that built that page and they just need to list apps pending approval. (Btw, I have zero memory on having to wait for approval for my app so this might be a new thing) Once your app is approved then I expect it will show up in that screen.

**Comment by nateswart** [comment]
Hello everyone. To confirm, yes the keys won't show up under your manage apps section until they are approved. We are updating our security policies and working through the queue of key approvals, so we appreciate your patience as we get our queue caught up to current demand. 

If you already have a key and are just waiting for "full access" requests to be processed then your keys should still be operational. Only brand new keys that haven't been approved at all yet are non-functional while they away review.

**Comment by jftanner** [comment]
@nateswart Can you provide some kind of ETA or tracking for this for new apps being approved?

Also, because it looks like the app just was never created (rather than is in a pending state), I assumed there was something wrong with my entries and tried three times. I only actually need one of them. I suspect I'm not alone in this, so that may well be leading to an increased queue that need review. Adding a "pending" listing on the "Manage your apps" would probably go a great way towards fixing that problem.

**Comment by nateswart** [comment]
Yes, we have noticed the same trend (it's ok, don't worry) :)

We've been ramping up a renewed attention for Open API, and a lot of changes are coming this year that will be yielding many improvements. The recommendation you mentioned is actually already in discussion with a number of other improvements. 

I'm hesitant to give any concrete timeframe that could cause further frustration for anyone where, but when I say "soon" know that I'm talking about days, not months.

Thank you for your patience and feedback!

**Comment by jftanner** [comment]
Thanks @nateswart! I'm really glad that the your APIs are going to be getting more love soon.

I'm a microservices architect and developer in my day job (selling craft goods is a side business for me), so a robust set of APIs is incredibly valuable for me. If I can build proper integrations between my inventory/finance systems and Etsy, that'll cut down on a lot of frustrating, manual effort as my business grows.

If you're looking for beta-testers or user feedback, please let me know how to sign myself up.

**Comment by ChrSchwarz** [comment]
It is still not working. Creating an app, status is pending and a view hours later it is gone in the managed apps.

**Comment by cdatwood** [comment]
I"m also experiencing a variation of this issue. I was able to create an app, it shows up in "Manage Apps," but whenever I try to use a public endpoint, I get an error that the API key is invalid. It's been a few weeks since the key was generated, I'd have thought it would work by now. 

URL: https://api.etsy.com/v3/application/openapi-ping
cURL: 
curl --location --request GET 'https://api.etsy.com/v3/application/openapi-ping' \
--header 'x-api-key: {{api key from Manage Apps}}' \
--header 'Cookie: user_prefs=Bxp6Y3HDL8zwXj6DXbKdxt0pfFpjZACCJO_wwzA6Oq80J0eHPCKWAQA.'

**Comment by ramo-karahasan-riechardt** [comment]
Hello,

we're also facing the same issue. @nateswart  is there any timeline you can give us or any alternative communication path we could take to continue developing against the API and testing it out?

**Comment by rakeshdobariya1** [comment]
> Hello everyone. To confirm, yes the keys won't show up under your manage apps section until they are approved. We are updating our security policies and working through the queue of key approvals, so we appreciate your patience as we get our queue caught up to current demand.
> 
> If you already have a key and are just waiting for "full access" requests to be processed then your keys should still be operational. Only brand new keys that haven't been approved at all yet are non-functional while they await review.

Now its "months". Still our application is pending approval. I request to reject app submitted for production review (Which we actually did not) , so at least we can carry on development. 

I think this ticket should be reopened as issue still exists ? 

Any words of hope ?

**Comment by rakeshdobariya1** [comment]
> Yes, we have noticed the same trend (it's ok, don't worry) :)
> 
> We've been ramping up a renewed attention for Open API, and a lot of changes are coming this year that will be yielding many improvements. The recommendation you mentioned is actually already in discussion with a number of other improvements.
> 
> I'm hesitant to give any concrete timeframe that could cause further frustration for anyone here if it moves at all, but when I say "soon" know that I'm talking about days, not months.
> 
> Thank you for your patience and feedback!

Now its "months". Still our application is pending approval. I request to reject app submitted for production review (Which we actually did not) , so at least we can carry on development.

I think this ticket should be reopened as issue still exists ?

Any words of hope ?

**Comment by ArtFiberStitch** [comment]
Well I can see this issue raised in April.  Guess what.  It is now August and the exact same thing is happening to me now.  Have applied about 5 times it just does not appear, if it comes, it says error and I apply again. There is definitely s problem.  One I am getting very frustrated with

**Comment by sdumper** [comment]
Why is this marked as closed it is not closed

**Comment by sdumper** [comment]
not requesting full access...but always shows this:
![image](https://user-images.githubusercontent.com/46971452/194893492-c4c0f53a-7654-47d9-aeaf-ef937251f124.png)

**Comment by Gareth064** [comment]
Guys, read #610 
If you're app has in anyway got to do with syncing Etsy data for the purpose of centralising it with data synced from other marketplaces... aka a Multichannel ecommerce app... then they aren't going to approve you're app registration request.

@ramo-karahasan-riechardt when you send the developers an email, you will get an automated response which makes it sound like they have "answered" and not going to do anything else about your email. But, all you have to do is respond to the automated email saying that nothing in the response helps you and you'd like someone to answer your query. This email will then go into a Zendesk service desk queue and a proper response should come from a human. 
You can also navigate to https://help.etsy.com/hc/en-gb/requests?segment=selling and raise a new ticket which will go straight into Zendesk and should end up with a response from a developer.

@sdumper it looks like you simply aren't giving Etsy enough information within the "Describe your application" section when requesting commercial access. Read the bits highlighted by the square box carefully.
![image](https://user-images.githubusercontent.com/22307223/194952718-81f6dd27-ee1c-498d-9e4d-afecd4733b10.png)

**Comment by ramo-karahasan-riechardt** [comment]
@Gareth064  thanks for your message. I am communicating with the folks at etsy,

![image](https://user-images.githubusercontent.com/1265056/195014035-f4418c98-7e71-4fea-b213-2acf4271f8d0.png)

That's what I got two month ago. Well, we just want to test out a private app... but it turns out, we're keeping getting Standard answers of catching up with App approvals and no new integrations will be approved.

I'm not sure what the rational behind this is, but it definetly is not positive for the Etsy dev community!

**Comment by katiecoathanger** [comment]
I still haven’t heard back from Etsy about this. Absolute worst service from tech support :(

**Comment by paramartad** [comment]
Maybe this is unrelated issue, but my app request keeps getting rejected. My use case is to list items from my own shop on my site (only shop_r, listings_r scope). No need for commercial access. Yet it keeps getting rejected? I have not even been able to use the "temporary" api key to get auth token to start developing/testing. 

IMO, the process of generating API key should be easier than this. There's no reason why generating API key for own personal use (i.e. no commercial access) should need human approval. Compared to other platforms like Shopify, this has been a huge pain in the a**.

**Comment by Gareth064** [comment]
@paramartad this must be a recent change. I have created new apps on my account many times without any approval process. But like you have pointed out, now when I create a new App, it now says it's pending approval. Seems bonkers.

**Comment by beeirl** [comment]
@Gareth064 did you finally get approved?

I am stuck in "pending approval" since several weeks.

**Comment by etsyachristensen** [comment]
Somewhere around mid to late November, Etsy began accepting requests for apps in the Inventory and Multi shop management spaces. Those apps submissions were put on hold earlier this year for a number of reasons, but now that much of what went into that decision has been addressed internally, we are accepting applications again and, if they are descriptive and provide the correct information - and do not violate our terms - they should get approved.

If you still have an app sitting in pending status, please contact developers@etsy.com and provide your name, your app name, your url and ask for a review of the app.
